infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -42,51 +42,216 @@ param subnetId string
+      // "Storage volume is currently in R/O mode".
+      // See https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#website_override_sticky_diagnostics_settings.
+      WEBSITE_OVERRIDE_STICKY_DIAGNOSTICS_SETTINGS: 0
+
+      // This value puts the Function App filesystem into readonly mode and runs code from a ZIP deployment.
+      WEBSITE_RUN_FROM_PACKAGE: '1'
+
+      // This is set by the Azure ZIP deploy commands in the YAML pipeline, so it is included here to prevent any
+      // unnecessary changes when a code deploy is pushed out.
+      SCM_DO_BUILD_DURING_DEPLOYMENT: false
+
+      FUNCTIONS_EXTENSION_VERSION: '~4'
+      FUNCTIONS_WORKER_RUNTIME: functionAppRuntime
+      APPINSIGHTS_INSTRUMENTATIONKEY: applicationInsightsKey
+    })
+    stagingOnlySettings: {
+      SLOT_NAME: 'staging'
+      // When deploying to a Function App utilising a secured VNet to store its deployment files,
+      // unique file shares must be pre-generated and unique to each slot prior to deploying the
+      // Function App itself if we wish to use slot swapping.
+      //
+      // In conjunction with WEBSITE_CONTENTAZUREFILECONNECTIONSTRING, this property tells the
+      // Function App that the deployment code resides in this Storage account and within *this*
+      // file share.
+      //
+      // See the second paragraph of https://learn.microsoft.com/en-us/azure/azure-functions/functions-infrastructure-as-code?tabs=json%2Clinux%2Cdevops&pivots=premium-plan#secured-deployments.
+      //
+      // When slots are swapped, this fileshare configuration value will swap with the slots and thereby make the new
+      // code deployment location available to the other slot.
+      WEBSITE_CONTENTSHARE: fileShareName2
+    }
+    prodOnlySettings: {
+      SLOT_NAME: 'production'
+      // As above, this value is distinct from the initial staging slot equivalent and will swap to the other slot upon
+      // a slot swap operation.
+      WEBSITE_CONTENTSHARE: fileShareName1
+    }
+    tagValues: tagValues
+  }
+  dependsOn: [
+    fileShare1
+    fileShare2
+  ]
+}
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-06-01-preview' existing = {
+  name: keyVaultName
+}
+
+// Allow Key Vault references passed as secure appsettings to be resolved by the Data Processor Function App itself.
+resource dataProcessorFunctionAppKeyVaultAccessPolicy 'Microsoft.KeyVault/vaults/accessPolicies@2021-06-01-preview' = {
+  name: 'add'
+  parent: keyVault
+  properties: {
+    accessPolicies: [
+    {
+      tenantId: functionApp.identity.tenantId
+      objectId: functionApp.identity.principalId
+      permissions: {
+        secrets: [
+          'list'
+          'get'
+        ]
+      }
+    }
+    {
+      tenantId: functionApp.identity.tenantId
+      objectId: functionAppSlotSettings.outputs.stagingSlotPrincipalId
+      permissions: {
+        secrets: [
+          'list'
+          'get'
+        ]
+      }
+    }]
+  }
 }
```

Comment:
Ooh yeah, good shout - I hadn't spotted that we'd had that before :)

Reply:
Done. I've also made permissions for the policy to be configurable but default to read-only access to secrets, and to take an array of principal ids rather than specifying one by one.